source/agora/test/Flash.d

```diff
@@ -1924,6 +1924,10 @@ unittest
     network.clients[0].postTransaction(update_tx);
     assert(!network.clients[0].hasTransactionHash(update_tx.hashFull()));
 
+    // make the `charlie` node not to send `update tx 6` unintentionally
+    // with `allow_publish` set to `false`, which cause failure to this test
+    Thread.sleep(1.seconds);
+
```

Comment:
Isn't the real problem in the logic of the disabling publishing?

Reply:
> Isn't the real problem in the logic of the disabling publishing?

Right, it is. But I was wondering whether it's worth modifying the logic for the disabling publishing which is only used in tests. Let me check more. This PR could be approved besides that because the checking needs some time.

Reply:
I just worry we are hiding a problem which might be test logic or might be a real bug in the implementation.

Reply:
I agree. It's meaningful to analyze the code more and I will add your concerns to the issue page and close the issue after handling this issue.